src/common/DeepPartial.ts

```diff
@@ -1,9 +1,13 @@
+import { NativeObject } from "./NativeObject"
+
 /**
  * Same as Partial<T> but goes deeper and makes Partial<T> all its properties and sub-properties.
  */
 export type DeepPartial<T> =
     | T
-    | (T extends Array<infer U>
+    | (T extends NativeObject
+          ? T
+          : T extends Array<infer U>
```

Comment:
For consistency purposes with other code parts its better to write it as:

```ts
export type DeepPartial<T> =
    | T
    | (T extends Array<infer U>
          ? DeepPartial<U>[]
          : T extends Map<infer K, infer V>
          ? Map<DeepPartial<K>, DeepPartial<V>>
          : T extends Set<infer M>
          ? Set<DeepPartial<M>>
          : T extends Date
          ? T
          : T extends object
          ? { [K in keyof T]?: DeepPartial<T[K]> }
          : T)
```

Reply:
probably we need to add `Buffer` as well...

```ts
export type DeepPartial<T> =
    | T
    | (T extends Array<infer U>
          ? DeepPartial<U>[]
          : T extends Map<infer K, infer V>
          ? Map<DeepPartial<K>, DeepPartial<V>>
          : T extends Set<infer M>
          ? Set<DeepPartial<M>>
          : T extends Date
          ? T
          : T extends Buffer
          ? T
          : T extends object
          ? { [K in keyof T]?: DeepPartial<T[K]> }
          : T)
```

Reply:
Buffer doesn't exist in the web browser, by adding it we are going to get issues in browser most likely.

Reply:
You are correct, `Buffer` doesn't exist in the web browser, but there are several reasons you need to add it:

* we already use `Buffer` everywhere - so it something won't work in the browser - it won't work anyway
* we can't remove the `Buffer` because users will start complain about them missing this feature
* actually for the browser it won't be a problem because it's `type-level` issue. Users will have to install @types/node with the Buffer in it if they have issues.